MigrationTutorial/Services/RealmService.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.IO;
+using Realms;
+using MigrationTutorial.Migrations;
+using MigrationTutorial.Utils;
+using Realms.Schema;
+
+namespace MigrationTutorial.Services
+{
+    public class RealmService
+    {
+        private static ulong _schemaVersion = 0;
+
+        private static RealmConfiguration _conf;
+
+        public static Realm GetRealm() => Realm.GetInstance(_conf);
+
+        public static void Init(ulong schemaVersion)
+        {
+            if (_schemaVersion == 0)
+            {
+                var realmPath = GetPath();
+                _schemaVersion = schemaVersion;
+
+                if (File.Exists(realmPath) && schemaVersion == 1)
+                {
+                    try
+                    {
+                        Logger.LogDebug($"Since the realm already exists and the supplied schema version is 1, it's assumed you want to start from scratch.\n Deleting {realmPath}");
+                        File.Delete(realmPath);
```

Comment:
Would we want to use the API that we offer ourselves here instead?

```suggestion
                        DeleteRealm(_realmConfiguration);
```